### ISSUE TITLE
1.4.1 Kafka Performance Issue Analysis & Investigation - Implement findings and recommendations as of PI3.2 - Part2

### ISSUE THREAD

**Issue by elnyry** [mojaloop/project#386]
## **Goal**:
1. As a contributor
I would like to _implement the findings & recommendations from the Kafka Performance Issue Analysis & Investigation_
so that transfers can be performant
2. As a Switch
I would want _the development team to implement Kafka Performance Issue Analysis & Investigation findings & recommendations_
so that transfers can be performant

### Central-Ledger
- [x] ~Kafka Performance - Implement Configuration changes [ @mdebarros ]~
- [x] ~Kafka Performance - Implement config switch to enable/disable Auto-Commit and/or Manual-Commits for Central-Ledger [ @mdebarros - 20180801T14h30BST - 100% ]~
- [x] ~Kafka Performance - Implement increased Affinity capabilities  [ @mdebarros - 20180801T14h30BST - 100% ]~
- [x] ~Unit Tests for Central-Ledger  [ @mdebarros - 20180801T14h30BST - 100% ]~
- [x] ~Integration Tests for Central-Ledger  [ @mdebarros  - 20180802T13h20BST - 100%]~
- [x] Discuss proposed approaches with DA for a decision on the approach to be implemented [ @mdebarros - 100%]
- [x] Add cron-job to PositionHandler to register new FSPs that are onboared [ @mdebarros - 20180809 ]
- [x] Add cron-job to PrepareHandler to register new FSPs that are onboared [ @mdebarros - 20180809 ]
- [x] Kafka Performance - Update Helm Chart values configuration to ensure latest version of Kafka is used for deployments [ @mdebarros - 20180816 - 100% ]
- [x] Kafka Performance - Update Helm Chart to support increased Affinity of deployments [ @mdebarros - 20180816 - 100% ]
- [x] Kafka Performance - Adding initContainer for Kafka connectivity [ @mdebarros - 20180816 - 100% ]
- [x] Kafka Performance - Adding initContainer for MySQL connectivity [ @mdebarros - 20180816 - 100% ]

### ML-API-Adapter
- [x] ~Kafka Performance - Implement Configuration changes for Ml-API-Adapter  [ @mdebarros ]~
- [x] ~Kafka Performance - Implement config switch to enable/disable Auto-Commit and/or Manual-Commits for Ml-API-Adapter [ @mdebarros - 20180801T14h30BST - 100% ]~
- [x] ~Kafka Performance - Implement increased Affinity capabilities  [ @mdebarros - 2018080102T14h30BST - 100% ]~
- [x] ~Unit Tests for Ml-API-Adapter  [ @mdebarros - 20180802T16h30BST - 100% ]~
- [x] ~Integration Tests for Ml-API-Adapter [ @mdebarros - 20180802T16h30BST - 100% ]~
- [x] Kafka Performance - Update Helm Chart values configuration to ensure latest version of Kafka is used for deployments [ @mdebarros - 20180813 - 100% ]
- [x] Kafka Performance - Update Helm Chart to support increased Affinity of deployments [ @mdebarros - 20180813 - 100% ]
- [x] Kafka Performance - Adding readyness and liveness probes [ @mdebarros - 20180816 - 100% ]
- [x] Kafka Performance - Adding initContainer for Kafka connectivity [ @mdebarros - 20180816 - 100% ]

### Testing
- [x] Test full deployment with ml-api-adapter and central-ledger changes

## **Acceptance Criteria**:
- [x] ~Implemented increased Affinity capabilities from https://github.com/mojaloop/project/issues/364 into PI3 code-base for Central-Ledger~
- [ ] Implemented increased Affinity capabilities into Helm deployments for Central-Ledger
- [x] ~Implemented increased Affinity capabilities from https://github.com/mojaloop/project/issues/364 into PI3 code-base for Ml-API-Adapter~
- [x] Implemented increased Affinity capabilities into Helm deployments for Ml-API-Adapter
- [x] Implemented configurations from [Recommendations & Findings as of PI3.3](https://mdebarros.github.io/kafkapoc-results/20180728/) into Helm for Central-Ledger
- [x] ~Implemented configurations from [Recommendations & Findings as of PI3.3](https://mdebarros.github.io/kafkapoc-results/20180728/) into PI3 code-base for Central-Ledger~
- [x] Implemented configurations from [Recommendations & Findings as of PI3.3](https://mdebarros.github.io/kafkapoc-results/20180728/) into Helm for Ml-API-Adapter
- [x] ~Implemented configurations from [Recommendations & Findings as of PI3.3](https://mdebarros.github.io/kafkapoc-results/20180728/) into PI3 code-base for Ml-API-Adapter~

## **Pull Requests**:
- [x] https://github.com/mojaloop/central-ledger/pull/117
- [ ] https://github.com/mojaloop/helm/pull/67 [ 2018-08-16 ]

## **Dependencies**:
- [x] #371 

## **Accountability**:
- Owner: @mdebarros
- QA/Review: Design Authority, Dev-Team

**Comment by kjw000** [comment]
this does not look like it will address/improve performance any further.  Will there be another story to address multi-threading and batch processing?

**Comment by elnyry** [comment]
Hi Kim, the affinity aspect has already been implemented as part of the story #371 , the current story is for updating the helm scripts (to help with deployment configurations, etc)

**Comment by mdebarros** [comment]
Continuing on story after unblocking #392
